### PR TITLE
Add awesome-lint and CI to lint new awesome lists

### LINKED ISSUE #569: Tools for enforcing awesome-style?
In the [manifesto](https://github.com/sindresorhus/awesome/blob/master/awesome.md) there are a few suggestions and requirements.  For example..
##### style
- grammatically correct
- no typos
- no markdown errors
- consistent style (e.g. "[link](/thing) - Description with a period.")
##### build
- table-of-contents generated

---

Have any list maintainers written some scripts to either lint for style issues? Also interested in boring build scripts to automate TOC generation with DocToc, etc.

### LINKED ISSUE #1365: Suggestion: Add CI to automate heavy PR load
This would ideally combine [awesome-lint](https://github.com/sindresorhus/awesome-lint) with a linter for the pull requests themselves and the main `awesome` readme.

Related to #1363 and the goal of maintaining Sindre's sanity 💯 

Here's my proposed game plan:
- create a linter based off of `awesome-lint` that lints this repo's `readme.md` to verify formatting and extract all the sub-lists
- this linter validates every awesome list linked to via `awesome-lint`
- maintainers will only consider PRs once they pass this CI step

There will be some existing awesome lists that don't pass `awesome-lint`, so we could whitelist these for now and incrementally fix them until the whitelist can be removed.

This would also put more onus on the `awesome-lint` project which appears to have been inactive for some time.

Possible future extensions:
- Include `awesome-lint` in the default `awesome` list template such that it would be enforced via CI but sub-awesome lists going forwards.

@sindresorhus I'd be happy to spearhead this, since I really believe these lists help countless developers out there every day, and the current gameplan of manual reviewing (even with the help of fellow list-makers) seems sub-optimal.

### PR DESCRIPTION
**Pull request by transitive-bullshit**
This PR incorporates [awesome-lint](https://github.com/sindresorhus/awesome-lint) and is aimed at providing automated linting to ease the load of reviewing new PRs.

It also adds a `.travis.yml` file to enable [travis-ci](https://travis-ci.com/) testing for PRs.

Note that it is not our intention to fully automate PR reviews, as content is still very important and will always require a manual pass. Instead, we're trying to reduce the amount of time it takes to manually review and maintain the Awesome project.

For new and existing Awesome list authors, lists are not required to pass `awesome-lint` **yet**, but it is a good idea to run the linter on your list before submitting, as it will help you find a lot of common issues and will hopefully save us all a great deal of time in the long run.

For more background, see https://github.com/sindresorhus/awesome-lint/issues/39

cc @sindresorhus @itaisteinherz

Fixes #1365.
Fixes #569.
Helps with #1363.

(Edit) Note that this PR is dependent on https://github.com/sindresorhus/awesome-lint/pull/43

### CODE REVIEW

File: test/_whitelist.js [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+	'https://github.com/johnjago/awesome-theravada',
+	'https://github.com/inspectit-labs/awesome-inspectit',
+	'https://github.com/nayafia/awesome-maintainers',
+	'https://github.com/xxczaki/awesome-calculators',
+	'https://github.com/ZYSzys/awesome-captcha',
+	'https://github.com/markusschanta/awesome-jupyter',
+	'https://github.com/andrewda/awesome-frc',
+	'https://github.com/engagingspaces/awesome-humane-tech',
+	'https://creativecommons.org/publicdomain/zero/1.0/',
+	'https://sindresorhus.com/'
```
Comment (itaisteinherz) [inline test/_whitelist.js]: The last two links can be removed from the whitelist since they wouldn't be linted anyways.

Comment (transitive-bullshit) [reply test/_whitelist.js]: Good call 😄

**Comment by itaisteinherz** [comment]
@sindresorhus Would love to know your opinion on this and the last 4 comments in sindresorhus/awesome-lint#39.

**Comment by sindresorhus** [comment]
I'm a bit reluctant of merging this before https://github.com/remarkjs/remark/issues/198 is resolved. It's going to cause a lot of noise. I'm also not sure whether we should have the `unordered-list-marker-style` rule enabled. Many lists use different style and I don't think it's worth the churn to enforce that.

**Comment by sindresorhus** [comment]
> I'm a bit reluctant of merging this before remarkjs/remark#198 is resolved.

Doesn't look like it will move forward anytime soon. Could we just disable that rule here?

**Comment by davisonio** [comment]
Like this idea but I've found that some awesome-lint rules are wayy too restrictive. Would be nice if we could disable rules on a per-list basis that are problematic. For example some lists have icons before each awesome item which breaks awesome-lint (as well as my list which for some reason won't recognise the badge even though it's correct).

**Comment by itaisteinherz** [comment]
> as well as my list which for some reason won't recognise the badge even though it's correct

@davisonio It would be great if you could open a specific issue for this over at [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) :)

**Comment by sindresorhus** [comment]
Moving this to an issue as it has gotten stale: https://github.com/sindresorhus/awesome/issues/1619

**Pull request closed without merging**